### PR TITLE
Catalog CRD support

### PR DESCRIPTION
**Pull request by zreigz**
Catalog CRD support

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

**Comment by michaeljguarino** [comment]
We should be sure to add create bindings too w/ the schema change done here: https://github.com/pluralsh/console/pull/1458

File: go/controller/api/v1alpha1/catalog_types.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		attrs.Tags = make([]*console.TagAttributes, 0)
+		for k, v := range c.Spec.Tags {
+			attrs.Tags = append(attrs.Tags, &console.TagAttributes{
+				Name:  k,
+				Value: v,
+			})
+		}
+	}
+
+	if c.Spec.Bindings != nil {
```
Comment (michaeljguarino) [inline go/controller/api/v1alpha1/catalog_types.go:89]: Catalogs should also have create bindings too

File: go/controller/api/v1alpha1/catalog_types.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		attrs.ReadBindings = PolicyBindings(c.Spec.Bindings.Read)
+		attrs.WriteBindings = PolicyBindings(c.Spec.Bindings.Write)
```
Comment (michaeljguarino) [inline go/controller/api/v1alpha1/catalog_types.go:91]: will this re-enqueue if the bindings aren't resolved?

Comment (zreigz) [reply go/controller/api/v1alpha1/catalog_types.go:91]: yes, in the controller:
```
	// Sync Catalog CRD with the Console API
	if err := r.ensure(catalog); err != nil {
		if goerrors.Is(err, operrors.ErrRetriable) {
			utils.MarkCondition(catalog.SetCondition, v1alpha1.SynchronizedConditionType, v1.ConditionFalse, v1alpha1.SynchronizedConditionReasonError, err.Error())
			return requeue, nil
		}
		return ctrl.Result{}, err
	}
```

File: go/controller/api/v1alpha1/catalog_types.go (hunk 1 above)
Comment (zreigz) [reply go/controller/api/v1alpha1/catalog_types.go:89]: The create binding is missing in the newest schema.graphql

Comment (michaeljguarino) [reply go/controller/api/v1alpha1/catalog_types.go:89]: it should be present here: https://github.com/pluralsh/console/pull/1458

File: go/controller/api/v1alpha1/catalog_types.go (hunk 2 above)
Comment (michaeljguarino) [reply go/controller/api/v1alpha1/catalog_types.go:91]: Oh i see cool

File: go/controller/api/v1alpha1/catalog_types.go (hunk 1 above)
Comment (zreigz) [reply go/controller/api/v1alpha1/catalog_types.go:104]: done

**Pull request merged**